controllers/sysbench/job.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2019 The xridge kubestone contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sysbench
+
+import (
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/firepear/qsplit"
+	perfv1alpha1 "github.com/xridge/kubestone/api/v1alpha1"
+)
+
+// NewJob creates a sysbench benchmark job
+func NewJob(cr *perfv1alpha1.Sysbench) *batchv1.Job {
+	labels := map[string]string{
+		"app":               "sysbench",
+		"kubestone-cr-name": cr.Name,
+	}
+	for key, value := range cr.Spec.PodLabels {
+		labels[key] = value
+	}
+
+	sysbenchCmdLineArgs := []string{}
+	sysbenchCmdLineArgs = append(sysbenchCmdLineArgs, qsplit.ToStrings([]byte(cr.Spec.Options))...)
+	sysbenchCmdLineArgs = append(sysbenchCmdLineArgs, cr.Spec.TestName, cr.Spec.Command)
+
+	backoffLimit := int32(0)
+
+	job := batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name,
+			Namespace: cr.Namespace,
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "sysbench",
+							Image:           cr.Spec.Image.Name,
+							ImagePullPolicy: corev1.PullPolicy(cr.Spec.Image.PullPolicy),
+							Command:         []string{"/usr/local/bin/sysbench"},
+							Args:            sysbenchCmdLineArgs,
+						},
+					},
+					ImagePullSecrets: []corev1.LocalObjectReference{
+						{
+							Name: cr.Spec.Image.PullSecret,
+						},
+					},
+					RestartPolicy: corev1.RestartPolicyNever,
```

Comment:
Besides Affinities, we should include here CPU and Memory limits.
It is fine to include that in an other PR, but then please add a // TODO item so that it is not forgotten.

Reply:
Done.